docs/core-concepts/app-deployment.md

```diff
@@ -1,17 +1,255 @@
+
+## Why Deploy a Portable App? ##
+
+Deploying a portable app has a number of advantages:
+
+- You do not have to define the target operating systems that your .NET Core app will run on in advance. Because .NET Core uses a common PE file format for executables and libraries regardless of operating system, .NET Core can execute your app regardless of the underlying operating system.
+
+- The size of your deployment package is small. You only have to deploy your app and its dependencies, not .NET Core itself.
+
+- Multiple apps use the same .NET Core installation, which reduces both disk space and memory usage on host systems.
+
+There are also a few disadvantages:
+
+- Your app can run only if the version of .NET Core that you target, or a later version, is already installed on the host system. 
+
+- It is possible for the .NET Core runtime and libraries to change without your knowledge. In rare cases, this may change the behavior of your app.
+
+## Deploying a Simple Portable App ##
+
+Deploying a portable app with no third-party dependencies simply involves building, testing, and publishing the app. 
+A simple example written in C# illustrates the process. The example uses the dotnet utility from the command line; however, 
+you can also use a development environment, such as Visual Studio or Visual Studio Code, to compile, test, and publish the example.
+
+1. Create a directory for your project, and from the command line, type `dotnet new` to create a new C# console project. 
+
+2. Open the `Program.cs` file in an editor, and replace the auto-generated code with the following code. It prompts the user to enter text, and then displays the individual words entered by the user. It uses the regular expression `\w+` to separate the words in the input text.
+
+    ```cs
+    using System;
+    using System.Text.RegularExpressions;
+
+    namespace Applications.ConsoleApps
+    {
+        public class ConsoleParser
+        {
+            public static void Main()
+            {
+                 Console.WriteLine("Enter any text, followed by <Enter>:\n");
+                 String s = Console.ReadLine();
+                 ShowWords(s);
+                 Console.Write("\nPress any key to continue... ");
+                 Console.ReadKey();
+          }
+
+          private static void ShowWords(String s)
+          {
+              String pattern = @"\w+";
+              var matches = Regex.Matches(s, pattern);
+              if (matches.Count == 0)
+                  Console.WriteLine("\nNo words were identified in your input.");
+              else {
+                  Console.WriteLine("\nThere are {0} words in your string:", matches.Count);
+                  for (int ctr = 0; ctr < matches.Count; ctr++)
+                      Console.WriteLine("   #{0,2}: '{1}' at position {2}", ctr,
+                                        matches[ctr].Value, matches[ctr].Index);
+              }
+          } 
+      }
+    }
+    ```

```

Comment:
* This code has inconsistent indentation. Also, in C#, it's more common to put each brace on a new line.
* Wouldn't it be better to use either `Write` and `\n` or just `WriteLine`, instead of both `WriteLine` and `\n`?
* Why is the loop variable named `ctr`? I would use either the common `i`, or expand the name to be clear (`counter`?).